doc/sphinx/proofs/writing-proofs/reasoning-inductives.rst

```diff
@@ -5,269 +5,289 @@ Reasoning with inductive types
+     Giving multiple :n:`@induction_clause`\s is equivalent to applying :n:`destruct`
+     serially on each :n:`@induction_clause`.
 
-      This behaves like :n:`destruct @term` providing explicit instances for
-      the dependent premises of the type of :token:`term`.
+   :n:`@natural`
+     Equivalent to :tacn:`intros` :n:`until @natural` followed by `destruct` applied to
+     the last introduced hypothesis.
 
-   .. tacv:: edestruct @term
-      :name: edestruct
+     .. note::
+        To destruct a number, use the syntax :n:`destruct (@natural)` (not
+        very interesting anyway).
 
-      This tactic behaves like :n:`destruct @term` except that it does not
-      fail if the instance of a dependent premises of the type
-      of :token:`term` is not inferable. Instead, the unresolved instances
+   :n:`@ident` (in :n:`@induction_arg`)
+     If :token:`ident` is a quantified variable in the conclusion
+     of the goal, then :n:`destruct @ident` behaves like :tacn:`intros`
+     :n:`until @ident; destruct @ident`.
+     If :token:`ident` is a hypothesis of the context and :token:`ident`
+     is no longer dependent in the goal after application
+     of :tacn:`destruct`, it is erased (to avoid erasure, use parentheses, as
+     in :n:`destruct (@ident)`).
+
+   :n:`as @or_and_intropattern`
+      Provides names for the variables introduced in each new subgoal.  The
+      :token:`or_and_intropattern` must have one :n:`{* @intropattern }`
+      for each constructor, given in the order in which the constructors are
+      defined.  If there are not enough names, Coq picks fresh names.
+
+   :n:`eqn : @naming_intropattern`
+      Generates a new hypothesis in each new subgoal that is an equality between
+      the term being case-analyzed and the associated constructor (applied to
+      its arguments).  The name of the new item may be specified in the
+      :n:`@naming_intropattern`.
+
+   :n:`with @bindings`  (in :n:`@one_term_with_bindings`)
+      Provides explicit instances for
+      the :term:`dependent premises <dependent premise>` of the type of
+      :token:`one_term`.
+
+   :n:`@occurrences`
+     Selects specific subterms of the goal and/or hypotheses to apply
+     the tactic to.  See :ref:`Occurrence clauses <occurrenceclauses>`.
+     If it occurs in the :n:`@induction_principle`, then
+     there can only be one :n:`@induction_clause`, which can't have its
+     own :n:`@occurrences` clause.
+
+   :n:`@induction_principle`
+     Makes the tactic equivalent to
+     :tacn:`induction` :n:`{+, @induction_clause } @induction_principle`.
+
+   .. tacn:: edestruct {+, @induction_clause } {? @induction_principle }
+
+      If the type of :n:`@one_term` (in :n:`@induction_arg`) has
+      :term:`dependent premises <dependent premise>` or dependent premises
+      whose values are not inferrable from the :n:`with @bindings` clause,
+      :n:`edestruct` turns them into existential variables to be resolved later on.
```

Comment:
This paragraph is sufficient alone. You can remove the next one.

Reply:
Whoops, right.